harness/features/track.local.test.ts

```diff
@@ -36,10 +35,8 @@ describe('Track Tests - Local', () => {
             url = getConnectionStringForProxy(sdkName)
 
             scope
-                .get(
-                    `/client/${client.clientId}/config/v1/server/${client.sdkKey}.json`,
-                )
-                .reply(200, config)
+                .get(client.getValidConfigPath())
+                .reply(200, client.getValidConfig)
```

Comment:
passing the function like this might have issues with binding to the `this` context of the client instance, like if this function is using the `this` keyword in it then it wont work

Reply:
wait what? can you explain more?

Reply:
basically in JS  a function is bound to the context its called from, so if you do
```client.someMethod()```
then within that method `this` refers to the client
but if you do something like 
```
const method = client.someMethod
method()
```
then when invoked `this` will be undefined because its not being called within the same context
to fix it you can either do:
```
const method = client.someMethod.bind(client)
method()
```

which makes it so that the method is always bound to "client"

or you can use an arrow function like

```
const method = () => client.someMethod()
method()
```
to basically "wrap" it while preserving context

you can also write the method as an arrow function on the class itself

```
class Client {
 method = () => {
}
}
```
which then makes it always use `this` of the client